### ISSUE TITLE
0.8.0: type-annotation-spacing bug

### ISSUE THREAD

**Issue by pat841** [nzakas/eslint-plugin-typescript#79]
In 0.7.0, this line would pass lint checking:

```javascript
  resolver: (() => PromiseLike<T>) | PromiseLike<T>;
```

In 0.8.0, I get this error:
>   13:13  error  Expected a space after the '('  typescript/type-annotation-spacing

where `(` in the error represents the first parenthesis after the `: `

**Comment by j-f1** [comment]
Thanks for reporting this!

Do these ones throw the error?

```ts
interface Foo<T> {
  resolver: (() => PromiseLike<T>);
  resolver: () => PromiseLike<T>;
  resolver: (PromiseLike<T>);
  resolver: () => PromiseLike<T> | PromiseLike<T>;
}
```

**Comment by pat841** [comment]
@j-f1

Using 0.8.0:
```javascript
interface Foo<T> {
  resolverOne: (PromiseLike<T>); // type-annotation-spacing error
  resolverTwo: (() => PromiseLike<T>); // type-annotation-spacing error
  resolverThree: () => PromiseLike<T>;
  resolverFour: () => PromiseLike<T> | PromiseLike<T>;
}
```
The issue with writing `() => PromiseLike<T> | PromiseLike<T>;` is because TypeScript assumes the union is for the arrow function result type and NOT the resolver type union.

**Comment by weirdpattern** [comment]
@j-f1 are you working on this?
I think I know what's going on and will be happy to submit a PR to get this fixed

**Comment by j-f1** [comment]
@weirdpattern I’m kinda busy, so go ahead!
